File: app/src/main/java/tutorials/acviewmodel/viewmodel/ViewModelModule.java
package tutorials.acviewmodel.viewmodel;

import android.arch.lifecycle.ViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import tutorials.acviewmodel.home.ListViewModel;
import tutorials.acviewmodel.home.SelectedRepoViewModel;

@Module
public abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(ListViewModel.class)
    abstract ViewModel bindListViewModel(ListViewModel viewModel);

    @Binds
    @IntoMap
    @ViewModelKey(SelectedRepoViewModel.class)
    abstract ViewModel bindSeelectedRepoViewModel(SelectedRepoViewModel viewModel);
}

File: app/src/main/java/technologies/acviewmodel/viewmodel/ViewModelModule.kt
package technologies.acviewmodel.viewmodel

import android.arch.lifecycle.ViewModel

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import technologies.acviewmodel.screens.repos.viewmodel.ListViewModel
import technologies.acviewmodel.screens.repos.viewmodel.MainActivityViewModel
import technologies.acviewmodel.screens.repos.viewmodel.SelectedRepoViewModel

@Module
abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(MainActivityViewModel::class)
    internal abstract fun bindMainViewModel(viewModel: MainActivityViewModel): ViewModel

    @Binds
    @IntoMap
    @ViewModelKey(ListViewModel::class)
    internal abstract fun bindListViewModel(viewModel: ListViewModel): ViewModel

    @Binds
    @IntoMap
    @ViewModelKey(SelectedRepoViewModel::class)
    internal abstract fun bindSelectedRepoViewModel(viewModel: SelectedRepoViewModel): ViewModel

}
